tests/pytorch/mock_sparse/test_transpose.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+import torch
+
+from dgl.mock_sparse import diag, create_from_coo
+
+
+@pytest.mark.parametrize("val_shape", [(3,), (3, 2)])
+@pytest.mark.parametrize("mat_shape", [None, (3, 5), (5, 3)])
+def test_diag_matrix_transpose(val_shape, mat_shape):
+    val = torch.randn(val_shape)
+    mat = diag(val, mat_shape).transpose()
+
+    assert torch.allclose(mat.val, val)
+    if mat_shape is None:
+        mat_shape = (val_shape[0], val_shape[0])
+    assert mat.shape == mat_shape[::-1]
+
+
+@pytest.mark.parametrize("dense_dim", [None, 2])
+@pytest.mark.parametrize("row", [[0, 0, 1, 2], (0, 1, 2, 4)])
+@pytest.mark.parametrize("col", [(0, 1, 2, 2), (1, 3, 3, 4)])
+@pytest.mark.parametrize("mat_shape", [(3, 5), (5, 3)])
+def test_sparse_matrix_transpose(dense_dim, row, col, mat_shape):
+    # Skip invalid matrices
+    if max(row) >= mat_shape[0] or max(col) >= mat_shape[1]:
+        return
```

Comment:
Perhaps it's better to replace `mat_shape` with `extra_shape` or `shape_offset` for expanding the shape dictated with `row` and `col`.

Reply:
I didn't get your point. Basically I think `mat_shape` cannot be inferred from `row` or `col`.

Reply:
Basically, we have

```
base_shape = (max(row) + 1, max(col) + 1)
target_shape = base_shape + offset_shape
```